### ISSUE TITLE
What to do with our old branches?

### ISSUE THREAD

**Issue by MekDrop** [ImpressCMS/impresscms#689]
We have many old branches that are not updated in many years. What do you we we could do with them? Is there anything there that we could reuse? Or we can just delete them?

| Branch | Last updated | What is this is about? | What I think |
|-----------|-------------------|------------------------------|------------------|
| feature/158-Admin_Login_Access | 9 years ago by Markus Schutz | | |
| feature/254-media_and_image_manager | 9 years ago by Markus Schutz | | |
| feature/XML_import_export | 9 years ago by Markus Schutz | Something about to export data as XML format | I can't think of reason for what to use it, but I think it would be not very hard to add this code to even 2.x releases line |
| feature/Yubikey_2Tier_Authentication | 9 years ago by Markus Schutz | Yubikey auth | Sounds good but I don't have any experience with it; also our auth system needs rework so I think if we really need this we probably need to redo this implementation | 
| feature/626-input_filtering | 8 years ago by Vaughan Montgomery | Input filtering | Properties basically does very similar thing; so I'm not sure if this is needed for us |
| feature/273-imcontrols | 8 years ago by MekDrop | IMControls - another front-end framework based on jQuery | I have invested  a lot of time into this but VueJS does some and works better; so I think we should use it instead |
| <del>features/online-core-update</del> | 8 years ago by fiammybe | Automatic core update | We have composer in 2.0 and I think we should use it for such functionality |
| feature/Controls_and_UI | 7 years ago by MekDrop | | |
| feature/restructure_system_module_with_actions | 7 years ago by MekDrop | System module use actions for everything | We now plan use 2.0 normal routes and controllers so I think this will be not usefull anymore |
| feature/2.0_UI-Polishing | 7 years ago by William Hall | | |
| feature/impresscms_2.0-installer | 7 years ago by skenow | Installer upgrade | |
| feature/actions | 7 years ago by MekDrop | Similar thing to controllers | We plan to use controllers so I think this will be not usefull anymore |
| branches/impresscms_2.0 | 7 years ago by fiammybe | Old ImpressCMS 2.0 branch | If I remember correctly there are some cool ideas for 2.0 GUI update, maybe we could port that into our current 2.0 |
| <del>feature/2.0-PDO</del> | 6 years ago by skenow | PDO support | We use now AuraSQL that has already good PDO support, so I don't think that this branch has any use |
| <del>feature/2.0-retro--smarty-uri-alfred</del> | 6 years ago by skenow | | |
| <del>branches/impresscms_1.2_LTS</del> | 4 years ago by skenow | 1.2.9 Final | We should tag this one and delete branch |
| dev/14-newtheme | 3 years ago by fiammybe | | |
| <del>fiammybe-link-contribute-to-contribute</del> | 2 years ago by fiammybe | Improved CONTRIBUTE.md | Why these changes stuck? |
| fix/TinyMCE-imagemanager-plugin-not-sanitizing-user-input | 2 years ago by fiammybe | | |
| libraries_1.4/css-tidy | 12 months ago by skenow | | I think this was merged into 1.4 |

**Comment by fiammybe** [comment]
I asked this question a long time ago (2 year I think) and it is still relevant. If we want, we can fork this repo in an archive repo of some sort, and cleanup the impresscms space.

**Comment by MekDrop** [comment]
I'm not sure if we need to keep them if we can't use. That's why my question: is here something that we should use?

**Comment by fiammybe** [comment]
# remove
* feature/158-Admin_Login_Access --> new login/authentication should be done first, and then this code will be obsolete
* feature/Yubikey_2Tier_Authentication --> new login/authentication library should make this code not useful
* features/online-core-update --> composer, as you said
* feature/2.0_UI-Polishing --> UI has evolved massively in the last 7 years
* feature/actions --> ok
* feature/2.0-PDO --> ok
* feature/restructure_system_module_with_actions -> sorry for all the work you did
* feature/273-imcontrols --> sorry for all the work you did, we never got a good documented example up and running i'm afraid
* feature/2.0-retro--smarty-uri-alfred --> this was a kind of routing, but we now have routes
* dev/14-newtheme --> themes will be done in their own repo
* fiammybe-link-contribute-to-contribute --> abracadabra : this disappeared (magic trick :-) )


# Keep
* feature/XML_import_export --> Being able to export the site data without needing to migrate the DB is a very interesting capability. Perhaps in a module, but best to keep the code.
* feature/254-media_and_image_manager

# Not sure
* branches/impresscms_2.0 --> we'll have to see if the stuff used isn't too old - frontend dev has evolved quite a bit
* fix/TinyMCE-imagemanager-plugin-not-sanitizing-user-input
* libraries_1.4/css-tidy

**Comment by skenow** [comment]
There is an archive feature in git, right?
> git tag archive/<branchname> <branchname>
> git branch -D <branchname>

**Comment by MekDrop** [comment]
@skenow good idea. Still I don't think so that we need to keep everything in this way. I think some, of this branches still exists on Assembla SVN. So, if somebody needs them probably they could get from there. Another thing that maybe some of these branches can't be useful at all... so I don't see any reason to keep them.

@fiammybe you can look in our repo tags list. There are few made in that way already.
